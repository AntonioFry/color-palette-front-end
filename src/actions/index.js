export const throwError = error => ({
  type: 'THROW_ERROR',
  payload: {
    error
  }
});

export const isLoading = isLoading => ({
  type: 'IS_LOADING',
  payload: {
    isLoading
  }
});

export const search = searchWord => ({
  type: 'SEARCH',
  payload: {
    searchWord
  }
});

export const addProject = project => ({
  type: 'ADD_PROJECT',
  payload: {
    project
  }
});

export const addPalette = palette => ({
  type: 'ADD_PALETTE',
  payload: {
    palette
  }
});

export const addProjects = projects => ({
  type: 'ADD_PROJECTS',
  payload: {
    projects
  }
});

export const addPalettes = palettes => ({
  type: 'ADD_PALETTES',
  payload: {
    palettes 
  }
});

export const removeProject = id => ({
  type: 'REMOVE_PROJECT',
  payload: {
    id
  }
});

export const removePalette = id => ({
  type: 'REMOVE_PALETTE',
  payload: {
    id
  }
});