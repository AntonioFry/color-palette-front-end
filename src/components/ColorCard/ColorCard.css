.color-card {
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
}

.card-hex-value {
  width: 100%;
  text-align: center;
}