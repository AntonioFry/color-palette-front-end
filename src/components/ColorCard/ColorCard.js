import React from 'react';

export const ColorCard = ({ hexValue }) => {
  return (
    <article>
      <h3>{hexValue}</h3>
      <button>Lock</button>
    </article>
  )
}