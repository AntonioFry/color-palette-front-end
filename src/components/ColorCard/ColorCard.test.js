import React from 'react';
import { ColorCard } from './ColorCard';
import { lockingColor } from '../../actions';
import { shallow } from 'enzyme';
import { mapDispatchToProps } from './ColorCard';

describe('ColorCard', () => {
  
  describe('ColorCard component', () => {
    let wrapper;
    let props;

    beforeEach(() => {
      props = {
        hexValue: '#ffffff',
        key: 1,
        id: 1
      }
      wrapper = shallow(<ColorCard hexValue={props.hexValue} key={props.key} id={props.id}/>)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    });

  });

  describe('mapDispatchToProps', () => {
    it('should dcall dispatch with currentColors', () => {
      const mockId = 1;
      const mockDispatch = jest.fn();
      const actionToDispacth = lockingColor(mockId);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.lockingColor(mockId)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispacth)
    })
  })

})