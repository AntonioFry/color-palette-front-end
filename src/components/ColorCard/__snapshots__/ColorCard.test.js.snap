// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorCard ColorCard component should match the snapshot 1`] = `
<div
  className="color-card"
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <p
    className="card-hex-value"
  >
    #FFFFFF
  </p>
  <i
    class="fas fa-lock-open"
    onClick={[Function]}
  />
</div>
`;
